datadog_checks_base/datadog_checks/base/checks/openmetrics/mixins.py

```diff
@@ -444,6 +428,28 @@ def reset_http_config(self):
         """
         self._http_handlers.clear()
 
+    def set_bearer_token(self, instance):
```

Comment:
I think calling this `instance` instead of `config` could lead to some confusion, as this is working on the config_map of an instance indirectly, not the instance itself.

It also seems like a lot of the work could be simplified / copied directly over if you change the function signature to be something like:
```
def set_bearer_token(self, instance, config):
```
where `config` is what you are currently calling instance, and `instance` is the instance that is also passed into `create_scraper_configuration`

You could also then move the old line, `bearer_token_auth = _get_setting('bearer_token_auth', False)` into the `set_bearer_token` function, something like:
`bearer_token_auth = instance.get('bearer_token_auth', default_instance.get('bearer_token_auth', False))`

Reply:
Oooh, great call! Changing this now.